pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go

```diff
@@ -591,17 +616,27 @@ func (sip *streamIngestionProcessor) bufferKV(kv *roachpb.KeyValue) error {
 }
 
 func (sip *streamIngestionProcessor) bufferCheckpoint(event partitionEvent) error {
-	log.Infof(sip.Ctx, "got checkpoint %v", event.GetResolved())
-	resolvedTimePtr := event.GetResolved()
-	if resolvedTimePtr == nil {
-		return errors.New("checkpoint event expected to have a resolved timestamp")
+	resolvedSpans := event.GetResolvedSpans()
+	if resolvedSpans == nil {
+		return errors.New("checkpoint event expected to have resolved spans")
 	}
-	resolvedTime := *resolvedTimePtr
 
-	// Buffer the checkpoint.
-	if lastTimestamp, ok := sip.bufferedCheckpoints[event.partition]; !ok || lastTimestamp.Less(resolvedTime) {
-		sip.bufferedCheckpoints[event.partition] = resolvedTime
+	lowestTimestamp := hlc.MaxTimestamp
+	highestTimestamp := hlc.MinTimestamp
+	for _, resolvedSpan := range resolvedSpans {
+		if resolvedSpan.Timestamp.Less(lowestTimestamp) {
+			lowestTimestamp = resolvedSpan.Timestamp
+		}
+		if highestTimestamp.Less(resolvedSpan.Timestamp) {
+			highestTimestamp = resolvedSpan.Timestamp
+		}
+		_, err := sip.frontier.Forward(resolvedSpan.Span, resolvedSpan.Timestamp)
+		if err != nil {
+			return errors.Wrap(err, "unable to forward checkpoint frontier")
+		}
 	}
+	log.Infof(sip.Ctx, "got checkpoint with %v spans and timestamps within [%v, %v]", len(resolvedSpans), lowestTimestamp, highestTimestamp)
```

Comment:
This will probably be too much logging;
at the very least log.V(1);
better yet: log.EveryN
even beterrer add metrics.

Reply:
I didn't like the idea of an arbitrary N so I just made `EarliestFrontierSpan`, `LatestFrontierSpan`, and `FrontierSpanCount` gauge metrics